awx/main/tasks/callback.py

```diff
@@ -98,8 +99,12 @@ def event_handler(self, event_data):
             host = event_data.get('event_data', {}).get('host', '').strip()
             if host:
                 event_data['host_name'] = host
-                if host in self.host_map:
-                    event_data['host_id'] = self.host_map[host]
+                if hasattr(self, 'constructed_backlinks'):
+                    if host in self.constructed_backlinks:
+                        event_data['host_id'] = self.constructed_backlinks[host]
```

Comment:
I'm going to walk-back this change. Firstly, the naming of the new data structure is impossible to understand.

This changed the prior behavior from (event) -> (constructed host id) to (event) -> (real host id). There's some tortured logic, but I don't agree with this. Best to link events back to the constructed host.

The reason we _might want_ to link the events to the real host is so that you could traverse from the host back to its events. Problem is, I don't believe this is a thing. Doing that would cross multiple partitions (as we painfully discovered in other issues), and that creates problems. I don't believe the API ever follows this relationship, although I will have to check.

With that established, linking to the constructed host actually gives more information because it's trivial to walk back to the real host, but going from the real host to the constructed host is much harder.